### ISSUE TITLE
Tray Icon

### ISSUE THREAD

**Issue by smaragdus** [navhaxs/BrightnessTray#9]
When put in the hidden part of the system tray, the program icon is (almost) invisible, screen:

![BrightnessTray 1 2 4 0 - 2019-05-18 - 003](https://user-images.githubusercontent.com/9025757/57968766-5673a780-7977-11e9-8fe4-9d7a6d01749c.png)

**Off-topic**
File version is wrong, it shows **1.2.3.2** in stead of **1.2.4.0**.

**Question**
Does **BrightnessTray** save any settings? I did not find any in program folder, in AppData and in Windows Registry.

**Edit**
On first run **BrightnessTray** crashed. I can provide the DUMP files (**.dmp**) if needed,

**Comment by navhaxs** [comment]
Hey @smaragdus thanks for the report - appreciated.

I've never run into this issue before myself, as I like to keep BrightnessTray pinned to the tray, and on Windows 10 the system tray flyout background is coloured with the accent colour:

![image](https://user-images.githubusercontent.com/5162718/57970405-51891500-79c4-11e9-8091-c6d48977181f.png)

A possible solution for your case would be to render the tray icon digits with a thin black border, or to change the text colour entirely. If this were to be implemented, I'd also want to make the text render style some kind of user-configurable option. (BrightnessTray actually has zero configuration right now, apart from generating the `Autostart` shortcut link file in the startup directory.)

How do either of these suggestions sound to you?

I suppose this is worth fixing as there are still Windows 8.1 users today :)

**Note on future development:** I've read that the next Windows 10 release 1903 will include an in-built brightness slider (yay), so I don't plan on actively further developing BrightnessTray, but I will definitely be fixing any bugs.

And please do feel free to send the crash dump to navhaxs at hotmail dot com :)

**Comment by navhaxs** [comment]
I will fix the file version in the next release - I was being a little lazy :P

**Comment by smaragdus** [comment]
@navhaxs 

First, thank you for the immediate response. I always appreciate when developers are as friendly and helpful as you.

> A possible solution for your case would be to render the tray icon digits with a thin black border, or to change the text colour entirely.

You may add a border both to tray icon and to digits. Or use black background for white digits as it is in the bottom right tray icon ([Mem Reduct](https://github.com/henrypp/memreduct)) in my screen-shot.

> I'd also want to make the text render style some kind of user-configurable option.

This would be excellent but I suppose it would require too much work, especially having in mind that you don't plan to develop BrightnessTray further. Windows 10 may include a brightness slider but I doubt that it would have the Caffeine feature,

By the way I think I spotted a typo in tray commands- shouldn't it be [Caffeine](https://en.wiktionary.org/wiki/caffeine) in stead of `Caffiene`?

**screen:**
![BrightnessTray 1 2 4 0 - 2019-05-19 - 002](https://user-images.githubusercontent.com/9025757/57975467-1f7eaf80-79d2-11e9-9e8a-9633f39e451d.png)

`Caffiene` also appears in [BrightnessTray main page](https://github.com/navhaxs/BrightnessTray) under **Extra features**,

> (BrightnessTray actually has zero configuration right now, apart from generating the Autostart shortcut link file in the startup directory.)

This means that BrightnessTray is portable. If you choose to add some configuration please also add an option to save the configuration file in program folder (next to the executable)- thus the program will remain portable.

In the past I used to write short reviews about portable programs in a forum dedicated to portable freeware. Because I like your program (simple, tiny, portable, open source) I may review it in the future (no matter that I had some quarrels with some moderators). Please tell me which is the minimum .NET version required for BrightnessTray to run.

> I suppose this is worth fixing as there are still Windows 8.1 users today :)

I would rather switch to Linux than update to Windows 10.

> And please do feel free to send the crash dump to navhaxs at hotmail dot com :)

I still keep the dump file but I suppose that the problem was not in BrightnessTray but in my system (I do not have the latest .NET version installed by the way) so that it might not be worth wasting your time since BrightnessTray works just fine now. Yet I can upload the dump file and post a download link here.

**Comment by navhaxs** [comment]
Yeah I think having a `BrightnessTray.ini` config file would be good to allow some basic system tray customisation. And the program would still be portable.

> .NET version required for BrightnessTray to run.

.NET Framework 4.0

> Caffeine in stead of Caffiene?

Oops I'll fix that typo - I totally missed that.

**Comment by smaragdus** [comment]
@navhaxs 

> Yeah I think having a `BrightnessTray.ini` config file would be good to allow some basic system tray customisation. And the program would still be portable.

Excellent.

> .NET Framework 4.0

Thanks,

**Comment by navhaxs** [comment]
Hi @smaragdus, I've just added basic text color customisability in https://github.com/navhaxs/BrightnessTray/releases/tag/v1.2.5

**Comment by resistancelion** [comment]
> Hey @smaragdus thanks for the report - appreciated.
> 
> I've never run into this issue before myself, as I like to keep BrightnessTray pinned to the tray, and on Windows 10 the system tray flyout background is coloured with the accent colour:
> 
> ![image](https://user-images.githubusercontent.com/5162718/57970405-51891500-79c4-11e9-8091-c6d48977181f.png)
> 
> A possible solution for your case would be to render the tray icon digits with a thin black border, or to change the text colour entirely. If this were to be implemented, I'd also want to make the text render style some kind of user-configurable option. (BrightnessTray actually has zero configuration right now, apart from generating the `Autostart` shortcut link file in the startup directory.)
> 
> How do either of these suggestions sound to you?
> 
> I suppose this is worth fixing as there are still Windows 8.1 users today :)
> 
> **Note on future development:** I've read that the next Windows 10 release 1903 will include an in-built brightness slider (yay), so I don't plan on actively further developing BrightnessTray, but I will definitely be fixing any bugs.
> 
> And please do feel free to send the crash dump to navhaxs at hotmail dot com :)

I think it is better to take theme settings to render the text,
It will add  theme support to the program,

Also, there's some reasons why you should keep developing:
Windows system tray will not have ability to change color/brightness profiles, as i thought so.
This user-profiling will be some kind of simplifying interfaces (Volume control have it, so, why brightness doesn't have built-in adjustments)
There are many users, who are using WIndows XP / 8 / 8.1 nowadays, just because Windows 10 is misbehaving on poor quality hardware, such as Lenovo notebooks
